hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/GeneralHashJoinBuilder.java

```diff
@@ -98,17 +119,23 @@
                         orderedClauses.stream().map(e -> e.getValue().transform())
                 ).collect(toList());
         // A probable javac bug forced us to extract this variable
+        // and not using method reference
         Stream<JoinClause<?, T0, ?, ?>> joinClauses = orderedClauses
                 .stream()
                 .map(e -> e.getValue().clause())
-                .map(fnAdapter::adaptJoinClause);
+                .map(joinClause -> fnAdapter.adaptJoinClause(joinClause));
+        Stream<Boolean> whereNullNotAllowed = orderedClauses
```

Comment:
I tried it, you can safely inline this into the usage below. The reason that `joinClauses` stream had to be extracted does not appear in this case.

Reply:
Done